src/odemis/driver/test/xt_client_test.py

```diff
@@ -31,13 +31,19 @@
 from odemis import model
 
 from odemis.driver import xt_client
-from odemis.driver.xt_client import DETECTOR2CHANNELNAME, XTTKDetector
+from odemis.driver.xt_client import DETECTOR2CHANNELNAME
 from odemis.model import ProgressiveFuture, NotSettableError
 from odemis.util import test
 
 logging.basicConfig(level=logging.INFO)
 
-TEST_NOHW = (os.environ.get("TEST_NOHW", 0) != 0)
+# Accept three values for TEST_NOHW
+# * TEST_NOHW = 1: not connected to anything => skip most of the tests
+# * TEST_NOHW = sim: xtadapter/server_sim.py running on localhost
+# * TEST_NOHW = 0 (or anything else): connected to the real hardware
+TEST_NOHW = os.environ.get("TEST_NOHW", 0)  # Default to Hw testing
+if TEST_NOHW != "sim":
+    TEST_NOHW = (TEST_NOHW == "1")  # convert values other than "sim" to False/True
```

Comment:
This feels like your hiding things. If someone accidentally has typed "sym", `TEST_NOHW` will be set to False, which implies hardware testing.

Reply:
So, you mean it'd be better to say "anything different from 0" is NOHW testing? Or it should raise an error when it's neither sim 0 or 1? So far, in the other test cases, we consider anything different from 0 is NOHW testing.

Reply:
Maybe the double negations are messing with my mind, but the way I read this is that anything different from 0 is HW testing.
```
TEST_NOHW = "something-different"
TEST_NOHW = (TEST_NOHW == "1")  = ("something-different" == "1") = False
```

TEST_NOHW = False => TEST_HW = True

Anyway, I would say it should raise an error for anything different than sim, 0 or 1.

Reply:
Yes. Right now it says "anything different from 1 is normal hardware testing". That's opposite to every other test cases. I'll change it to raise an error if it's different from the 3 values.